chapters/prevent_duplicate_indexing/index.rst

```diff
@@ -1,49 +1,70 @@
 Prevent Indexing Duplicates
 ---------------------------
 
-When indexing documents, it is common to have duplicate documents received by the search system. One can either remove the duplicates before sending the duplicates to Jina or leave it to Jina for handling the duplicates.
+Summary
+#######
 
-To prevent indexing duplicates, one needs to add ``_unique`` for the ``uses_before`` option. For example,
+When indexing documents, it is common to have duplicate documents received by the search system. This document will show you how to prevent indexing duplicates in Jina.
 
-.. confval:: Python API
 
+Feature Description
+###################
+
+One can either remove the duplicates before sending the duplicates to Jina or leave it to Jina for handling the duplicates.
+
+Before you start
+################
+
+Make sure you are familiar with `Executor YAML Sytax <https://docs.jina.ai/chapters/yaml/executor.html>`_.
+
+
+Implementation
+##############
+
+Use ``_unique``
+***************
+
+To prevent indexing duplicates, one needs to add ``_unique`` for the ``uses_before`` option. In the following example, three Documents are indexed by the Flow. We use `on_done` argument to check how many Documents are indexed after.
+
+.. confval:: Python API
 
     .. highlight:: python
     .. code-block:: python
 
-        from jina.flow import Flow
-        from jina.proto import jina_pb2
+        from jina import Flow, Document
+        import numpy as np
 
-        doc_0 = jina_pb2.Document()
-        doc_0.text = f'I am doc0'
-        doc_1 = jina_pb2.Document()
-        doc_1.text = f'I am doc1'
+        doc_0 = Document(text='I am doc0', embedding=np.random.rand(10))
+        doc_1 = Document(text='I am doc1', embedding=np.random.rand(10))
+
+
+        def assert_only_two_docs_indexed(rsp):
+            assert len(rsp.index.docs) == 2
 
-        def assert_num_docs(rsp, num_docs):
-            assert len(rsp.IndexRequest.docs) == num_docs
 
         f = Flow().add(
             uses='NumpyIndexer', uses_before='_unique')
 
         with f:
             f.index(
                 [doc_0, doc_0, doc_1],
-                on_done=lambda rsp: assert_num_docs(rsp, num_docs=2))
+                on_done=lambda rsp: assert_only_two_docs_indexed(rsp))
 
-Under the hood, the configuration yaml file, :file:``executors._unique.yml``, under the :file:``jina/resources`` is used. The yaml file is defined as below
+Under the hood, the configuration yaml file, :file:``jina/resources/executors._unique.yml``, is used. The yaml file is defined as below
 
 
 .. confval:: YAML spec
 
     .. highlight:: yaml
     .. code-block:: yaml
-
-        !DocIDCache
+        !DocCache
```

Comment:
The YAML spce seems missing?
![image](https://user-images.githubusercontent.com/44033547/107387929-4b559900-6ac3-11eb-8f42-7fa5718788fd.png)